Docs: Glossary documentation reworked (#5500)
WeblateOrg/weblate: docs/user/glossary.rst

Glossary
========

Each project can have an assigned glossary as a shorthand for storing

> terminology in all languages. Consistency is more easily maintained this way.

Author: Are they stored together though?
I thought this was just the new style of component or project level glossaries.

> terminology in all languages. Consistency is more easily maintained this way.

Reviewer: AFAIK the backing storage was changed: It was purely database-stored as a dedicated django entity (pardon my term misuse if any). But now it's actual components which are stored as `*.tbx` files.

Although, you are right... It's not correct to say that they are "stored together", rather that they are "represented together", but I couldn't pick right words for that... I meant that e.g. you easily can see what a term is translated to one language and simply jump to another and so on...

> terminology in all languages. Consistency is more easily maintained this way.

Reviewer 2: If you come to a solution, use it, please :)

.. image:: /images/glossary-tools.png

> Untranslatable terms

Author: Oh yea, that is right.

> Untranslatable terms

Author: ```suggestion
Not translatable terms
```

> Untranslatable terms

Reviewer: It's only one instance, for mouse-over text: https://hosted.weblate.org/translate/weblate/application/en/?checksum=6049b99cdd1c207a

But `Untranslatable` is not used at all... I'm ok if it will be changed in the UI thou.

> Untranslatable terms

Reviewer 2: closing as outdated

++++++++++++++++++++

.. versionadded:: 4.5

> Flagging certain glossary term translations ``read-only`` manually, or

Author: No, manually is clicking the sidebar and typing in "read-only". Or by bulk-editing.
I didn't really go into it, because if you are editing manually, you should know.
I might be wrong there, but this keeps it easy on the ones that know nothing.

> Flagging certain glossary term translations ``read-only`` manually, or

Reviewer: hmmm... I definitely wouldn't call a "bulk-edit" a "manual" way... especially in comparison with pressing a button in the UI for a specific term...

For ones who know nothing it's a bad form: it's suggests that there is another way to do it and they will have to rush through all other docs to find, what it is and if they need it... yet, I can't see an easy wayto put in words without rephrasing the whole paragraph... probably to resemble more the previous form...

> Flagging certain glossary term translations ``read-only`` manually, or

Author: Changed, however the ordering might be strange now. More natural to promote the click-menu option first?

> Flagging certain glossary term translations ``read-only`` manually, or

Reviewer: yeah, probably so... But, I still like the original variant more; IMHO it's much more straight-forward: It describes the semantics of the feature and only then suggests how to use it...

I'd also avoid direct mention of `bulk-edit` it just looks out-of-place and just rather add a reference to a flag-related chapter of the doc...

> Flagging certain glossary term translations ``read-only`` manually, or

Reviewer 2: closing as outdated

> using :guilabel:`Tools` ↓:guilabel:`Mark as read-only` means they can not

Author: Should is the imperative of the translator, they can not, because "read-only renders it impossible to do so.

> using :guilabel:`Tools` ↓:guilabel:`Mark as read-only` means they can not

Reviewer: But it's up to translators whether they will follow whatever was defined in glossary or not... and also now it's up to translators whether to to define the flag or not...

> using :guilabel:`Tools` ↓:guilabel:`Mark as read-only` means they can not

Reviewer: i.e. add the `read-only` flag to the glossary => the term shouldn't be translated in the translation.

> using :guilabel:`Tools` ↓:guilabel:`Mark as read-only` means they can not

Author: I think you are thinking of "forbidden" for the should. "read-only" is for sure "can't".

Adding an entry of the glossary as read-only would only prevent it from being changed, or translated _in the glossary_. Glossaries are just components now.
(I am so tired I wrote "preven't" X))

> using :guilabel:`Tools` ↓:guilabel:`Mark as read-only` means they can not

Reviewer: I suppose you misunderstand the difference between `forbidden`, `not translatable` and *`read-only` for non-gloosary components*:
* `forbidden` is term with a given translation. And exactly this translation shouldn't be used e.g. `directory` shouldn't be translated as `folder`.
* `not translatable` is a term in glossary that shouldn't be translated in other translations at all. It is implemented via the `read-only` flag for a glossary, but unlike a normal component it could be removed or set by a translator. So `read-only` here is nothing more than an implementation notice. e.g. if `cat` (unix command) is excessively used through the text without any quotation you may want to add it as `not translatable`, to keep others from translating it as `cat` (small animal of felis genus). 
* *`read-only` for non-glossary* is what you describe here: a translation that can't be changed by a translator.

This section should describe the `not translatable` semantics at the first place (as it primary aimed for users). And it's backing implementation in the second. That's what actually was pretty straight forward in previous version.

> using :guilabel:`Tools` ↓:guilabel:`Mark as read-only` means they can not

Author: I was saying it makes sense to describe "forbidden" as should not, and "read-only" as can not. Seeing as one is a possible error to avoid, and the other is impossible to make.

We went from "not meant to be translated" and "should not change in translation" for _read-only_. That sounds a lot like change is possible to me.

Am I trying to describe "read-only" or "forbidden", and what am I doing wrong?

I don't particularly like the introduction of "not translatable", as a term, because it isn't a flag.

> using :guilabel:`Tools` ↓:guilabel:`Mark as read-only` means they can not

Reviewer: > Am I trying to describe "read-only" or "forbidden", and what am I doing wrong?

> I don't particularly like the introduction of "not translatable", as a term, because it isn't a flag.

But "not translatable" just has a different semantics for glossaries, than "read-only" for normal components. It has to be introduced in some form.

All in all I believe it's time to call @nijel, and just ask to reimplement it via a new flag...

@nijel, long story short, implementation of not-translatable/untranslatable semantics through `read-only` flag looks quite confusing... Are there any reason not to do it via a separate flag?

> using :guilabel:`Tools` ↓:guilabel:`Mark as read-only` means they can not

Author: Why new flag? Take me through the current situation of what happens and what goes wrong.

If it has a function that is clearly described, ~ Can't be replicated in translation, and both strings and glossaries are components, what is the confusion?

Read-only means ~unchanged form only.
Forbidden means ~avoid this.

> using :guilabel:`Tools` ↓:guilabel:`Mark as read-only` means they can not

Reviewer: Yes, glossaries are components now, but the `read-only` flag is treated differently and mean a different thing.

I've already wrote it above, as for now:

* `read-only` for non-glossaries: the translation of the string itself **can not** be changed by a user. (that's exactly, what you say, it's a strict **can not**)
* `read-only` for glossaries: the term is not-translatable, i.e. the term **should not** be (**is meant not to** be) changed while translating *other strings*, it's just makes it appear highlighted in the glossary sidebar, a user **can** (permitted, able and free to) translate it in other strings, but he **shouldn't**. As for the term itself, unlike non-glossaries a user **can** set and remove the `read-only` flag as he wish, which effectively allows him to translate the term itself.

> using :guilabel:`Tools` ↓:guilabel:`Mark as read-only` means they can not

Reviewer 2: @comradekingu @Fat-Zer 
Guys, please close these endless conversations, this is hell to merge.

> using :guilabel:`Tools` ↓:guilabel:`Mark as read-only` means they can not

Reviewer: @orangesunny, this conversation had stalled without a good solution... I should had probably file a separate issue on the problem  rather than try to call **nijel** directly...

> be translated. Use this for names or other terms that should not be changed in translation.

Author: I specifically don't like the "should" in that RFC. But yes, that is exactly the case. Something bad will happen. It should maybe be "proper nouns", but I've tried to go a bit simpler on my style to get it past review lately.

This isn't the avoided one, this is the ~keep-same.

> be translated. Use this for names or other terms that should not be changed in translation.

Author: ```suggestion
be translated. Use this for brand names or other terms that should not be changed in translation.
```

> be translated. Use this for names or other terms that should not be changed in translation.

Reviewer 2: closing as outdated

.. versionadded:: 4.5

Flagging certain glossary term translations ``read-only`` by bulk-editing, typing in the flag, or

> by using :guilabel:`Tools` ↓:guilabel:`Mark as read-only` means they can not

Reviewer 3: This doesn't render well, see https://github.com/WeblateOrg/weblate/discussions/5857. Fixed in 0fa425e2f94aec7aebfdc4234e01d584309a8607

Reviewer 2: Thank you!

.. versionadded:: 4.5

> Flagging certain glossary term translations as ``forbidden``, manually or

Author: The comma is wrong here, but what do you mean?

> Flagging certain glossary term translations as ``forbidden``, manually or

Reviewer: I meant, same problem as for `read-only` above with `or` and `manually` ...

> Flagging certain glossary term translations as ``forbidden``, manually or

Reviewer 2: closing as outdated

.. versionadded:: 4.5

> Flagging certain glossary terms as ``terminology`` manually, or by

Reviewer 2: closing as outdated

> using :guilabel:`Tools` ↓:guilabel:`Mark as terminology` puts them in all
> glossary languages. Use this for terms that should be translated consistently.

Reviewer: I'd go with something like this:

```suggestion
using :guilabel:`Tools` ↓:guilabel:`Mark as terminology` adds them to all 
the languages in the glossary. Use this for important terms which translation
to each and every language should be well thought and should have consistent meaning
across all of them .
```

<hr>

Also this one completely misses that this option is available only to privileged users.

Author: I think I changed this afterwards. Will have to check.

Reviewer: The second part is fine, but the section still lacks a note that it can be used only by privileged users by default (probably in a different paragraph).
And the same thing as for bulk-edit etc as above...

Reviewer 2: closing as outdated
